### PR TITLE
Added stations search doctrine

### PR DESCRIPTION
**Pull request by ya-web**

### CODE REVIEW

File: src/Repository/StationRepository.php [hunk 1]
```diff
@@ -103,6 +103,79 @@ public function findAllPaginatedOrderedStations(int $page = 1, int $limit = 11,
...
+        $qb = $this->createQueryBuilder('s');
+
+        $alias = 's';
+        if ('displayName' === $searchKey) {
+            $qb->leftJoin(User::class, 'u', Expr\Join::WITH, 's.user = u.id');
+            $alias = 'u';
+        }
+        if (
+            'department' === $searchKey &&
+            is_numeric(substr($searchTerm, 0, 1))
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Repository/StationRepository.php:117]: substr($searchTerm, 0, 1) to take Corsica into account

File: src/Repository/StationRepository.php [hunk 2]
```diff
@@ -103,6 +103,79 @@ public function findAllPaginatedOrderedStations(int $page = 1, int $limit = 11,
...
+        ) {
+            $searchTerm = substr($searchTerm, 0, 2);
+        }
+
+        $searchResults = $qb->andWhere('lower('.$alias.'.'.$searchKey.') =:searchKey')
+            ->setParameter('searchKey', strtolower($searchTerm))
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Repository/StationRepository.php:123]: I wil recheck all occurence of searchTerm and searchKey, but searchTerm is for user provided term, and searchKey is for keywords extracted from searchTerm

File: src/Controller/StationsController.php [hunk 3]
```diff
@@ -85,6 +85,32 @@ public function mesStations(
...
+    /**
+     * @Route("/participer/stations/recherche", name="stations-search")
+     */
+    public function stationsSearch(
+        Request $request,
+        BreadcrumbsGenerator $breadcrumbsGenerator,
+        Search $searchService
+    ) {
+        $station = new Station();
+        $form = $this->createForm(StationType::class, $station, [
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Controller/StationsController.php]: done

File: src/Controller/StationsController.php [hunk 4]
```diff
@@ -85,6 +85,32 @@ public function mesStations(
...
+            'action' => $this->generateUrl('stations_new'),
+        ]);
+        $searchTerm = $request->request->get('stations-search');
+
+        if (!$searchTerm || '' === $searchTerm) {
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Controller/StationsController.php]: done

File: templates/pages/stations.html.twig [hunk 5]
```diff
@@ -5,42 +5,19 @@
...
+	{% include 'components/actionbar.html.twig' with {
 		actionbar: {
 			'classes': 'search',
 			'components': [
 				{
 					'name': 'side-block',
 					'data': {
 						'type': 'search',
-						'text': 'Ville, code postal...'
+						'text': 'Nom, ville, code postal...'
```
[the remark this answers is not in the document]
Comment (ya-web) [reply templates/pages/stations.html.twig]: done

File: templates/pages/stationsSearch.html.twig [hunk 6]
```diff
@@ -0,0 +1,70 @@
+{% extends 'base.html.twig' %}
+
+{% block main_content %}
+	{% include 'layout/header.html.twig' with ({
+		'title': 'Stations d’observation',
```
[the remark this answers is not in the document]
Comment (ya-web) [reply templates/pages/stationsSearch.html.twig]: done

File: src/Repository/StationRepository.php [hunk 7]
```diff
@@ -103,6 +103,79 @@ public function findAllPaginatedOrderedStations(int $page = 1, int $limit = 11,
...
+            $alias = 'u';
+        }
+        if (
+            'department' === $searchKey &&
+            is_numeric(substr($searchTerm, 0, 1))
+        ) {
+            $searchTerm = substr($searchTerm, 0, 2);
+        }
+
+        $searchResults = $qb->andWhere('lower('.$alias.'.'.$searchKey.') =:searchKey')
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Repository/StationRepository.php]: done

File: src/Repository/StationRepository.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Repository/StationRepository.php]: done

**Comment by kstefanini** [comment]
Nice refactoring!

**Review by kstefanini — APPROVED** [review]

**Pull request merged**